### PR TITLE
fix(alert): apply styling to disabled items

### PR DESCRIPTION
**Pull request by kelvindart**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?
`ion-select-option[disbled]` have no reflective styling, so visibly do not look any different to non-disabled options.

Issue Number: #18542

## What is the new behavior?
- Disabled options now have an opacity of 0.3.
- This is applied via targeting `[disabled] .alert-inner-button`

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

*Without styling:*
ion-select-option no styling

*With styling:*
ion-select-option no styling

### CODE REVIEW

File: core/src/components/alert/alert.scss [hunk 1]
```diff
@@ -150,6 +150,16 @@
   height: 100%;
 }
 
+
+// Alert Button: Disabled
+// --------------------------------------------------
+
+[disabled] .alert-button-inner {
+  cursor: default;
+  opacity: .3;
```
Comment (pranjaljately) [inline core/src/components/alert/alert.scss]: Please use `opacity: .5;` as it is the style commonly used for other disabled buttons.

Comment (kelvindart-certua) [reply core/src/components/alert/alert.scss]: Done 🙂

File: core/src/components/select/test/single-value/index.html [hunk 2]
```diff
@@ -133,6 +133,16 @@
         </ion-select>
       </ion-item>
 
+      <ion-item>
```
Comment (liamdebeasi) [inline core/src/components/select/test/single-value/index.html]: The `Statuses` alert select looks like it already has a disabled button in its test, so not sure if this test gives us any additional coverage.

**Comment by kelvindart-certua** [comment]
Hey @liamdebeasi - good point. I think I added it to test both the attribute _and_ the property. Shall I remove it?

**Comment by liamdebeasi** [comment]
Yeah probably best to remove it. At this point, I think that would just be testing Stencil's functionality.

**Comment by kelvindart-certua** [comment]
Awesome - I shall do. Thanks! :)

**Comment by kelvindart-certua** [comment]
Hey @liamdebeasi - I've reverted the unrequired tests. Hopefully ready to merge now 🙂. Thanks!

**Comment by kelvindart** [comment]
Hey @liamdebeasi - I've reverted the commit. Hopefully ready to merge now 🙂.

**Comment by liamdebeasi** [comment]
Thanks! We actually use classes to indicate disabled states to make it easier for users to customize, so I've updated your PR with that. Other than that, I think we're good to go!

**Comment by kelvindart-certua** [comment]
Hey @liamdebeasi - I saw this was merged a few days ago, was it included in 4.7.0? No problem if not, I'm just checking :). If it wasn't, any idea which version it will appear in?

**Comment by liamdebeasi** [comment]
@kelvindart-certua This has not been merged in yet. 4.7.0 had a fix for disabled items in `ion-datetime`. We hope to have this PR merged in soon.

**Comment by kelvindart-certua** [comment]
No worries - just seen `master` was merged into it (I misread it initially, which is why I was surprised it wasn't in `4.7.0` - thanks 😁.

**Comment by liamdebeasi** [comment]
Merged. Thank you!

**Pull request merged**